xml/System.Runtime.CompilerServices/Unsafe.xml

```diff
@@ -141,12 +172,22 @@
         <Parameter Name="elementOffset" Type="System.IntPtr" Index="1" FrameworkAlternate="dotnet-plat-ext-2.1;dotnet-plat-ext-2.2;net-5.0;net-6.0;netcore-3.0;netcore-3.1" />
       </Parameters>
       <Docs>
-        <typeparam name="T">The type of reference.</typeparam>
-        <param name="source">The reference to add the offset to.</param>
+        <typeparam name="T">The elemental type of the managed pointer.</typeparam>
+        <param name="source">The managed pointer to add the offset to.</param>
         <param name="elementOffset">The offset to add.</param>
-        <summary>Adds an element offset to the given reference.</summary>
-        <returns>A new reference that reflects the addition of offset to pointer.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Adds an element offset to the given managed pointer.</summary>
+        <returns>A new managed pointer that reflects the addition of offset to the source pointer.</returns>
```

Comment:
The pattern I've tried to follow throughout the doc is that I reserve _reference_ for instances where the API shape involves `T*` or `T&` and where the operation involves reading or writing a full _T_ at that address. You'll see this in APIs like `Copy`.

For APIs that perform arithmetic or comparisons against the pointer value rather than dereference it, I use _\[un\]managed pointer_, as this terminology matches the ECMA sections listed in the type's Summary section.

For APIs like `ReadUnaligned`, I take a little bit of license here. Nominally the overloads would've been `ReadUnaligned<T>(void*)` and `ReadUnaligned<T>(void&)`, which .NET fully supports but which C# disallows. In both of those cases they're taking void pointers, but they're not dereferencing a `void`. They're instead performing a read or write of some differently-sized data at that address. Since they're not dereferencing a `void` I opted to use the terminology _\[un\]managed pointer_ here rather than _reference_, since IMO it's more accurate.